### PR TITLE
NDEV-2252: Clean up api_client

### PR DESCRIPTION
**Pull request by andreisilviudragnea**
Related to https://github.com/neonlabsorg/neon-evm/pull/208.

- Enabled `serde_json` `"preserve_order"` feature so `StructLog` field order is preserved in JSON response bodies for debug calls.
- Removed redudant `Arc` wrappers and `.clone()` calls.
- Removed `#[allow(clippy::too_many_arguments)]` by creating the request object earlier in the call stack.
- Removed redundant `#[allow(unused)]`.

### CODE REVIEW

File: neon-tracer/src/api_client/client.rs [hunk 1]
```diff
@@ -24,31 +15,25 @@ use reqwest::{
...
 };
 
 use super::errors::NeonAPIClientError;
 
-#[derive(Clone)]
 pub struct Client {
-    config: Arc<Config>,
     pub neon_api_url: String,
-    pub http_client: Arc<ReqwestClient>,
+    pub http_client: ReqwestClient,
```
Comment (stanislav-tkach) [inline neon-tracer/src/api_client/client.rs:29]: 👍 I also noticed that `reqwest::Client` has an inner `Arc`, so there is no sense to wrap it once again.

File: neon-tracer/src/data_source/mod.rs [hunk 2]
```diff
@@ -30,29 +31,34 @@ pub const ERR: fn(&str, id: u64) -> Error = |e: &str, id: u64| -> Error {
...
 pub struct DataSource {
     tracer_db: TracerDb,
-    web3: Arc<Web3<Http>>,
+    web3: Web3<Http>,
     pub neon_api: NeonAPIDataSource,
     pub request_id: Arc<AtomicU64>,
     evm_metadata_account: Pubkey,
     pub rpc_client: Arc<RpcClient>,
+    pub config: Arc<Config>,
+    pub steps_to_execute: u64,
```
Comment (stanislav-tkach) [inline neon-tracer/src/data_source/mod.rs:40]: Can we just use the `NUM_STEPS_TO_EXECUTE` constant instead of a parameter?

**Review by stanislav-tkach — APPROVED** [review]

File: neon-tracer/src/service/geth.rs [hunk 3]
```diff
@@ -70,3 +57,30 @@ impl GethTraceServer for DataSource {
...
+
+impl From<TransactionArgs> for TxParamsRequestModel {
+    fn from(transaction_args: TransactionArgs) -> Self {
+        TxParamsRequestModel {
+            sender: transaction_args.from.unwrap_or_default(),
+            contract: transaction_args.to,
+            data: transaction_args.input.map(|v| v.0),
+            value: transaction_args.value,
+            gas_limit: transaction_args.gas,
+            // access_list: transaction_args.access_list,
```
[the remark this answers is not in the document]
Comment (andreisilviudragnea) [reply neon-tracer/src/service/geth.rs:69]: I added the argument back as it is present in the Geth struct too.

File: neon-tracer/src/data_source/mod.rs (hunk 2 above)
Comment (andreisilviudragnea) [reply neon-tracer/src/data_source/mod.rs]: Done

**Pull request merged**